cmd/buildkitd/main.go

```diff
@@ -639,6 +639,9 @@ func getListener(addr string, uid, gid int, tlsConfig *tls.Config) (net.Listener
 		if tlsConfig != nil {
 			bklog.L.Warnf("TLS is disabled for %s", addr)
 		}
+		if proto == "npipe" {
+			return getLocalListener(listenAddr)
+		}
 		return sys.GetLocalListener(listenAddr, uid, gid)
```

Comment:
@gabriel-samfira had also created `getLocalListener` for UNIX https://github.com/moby/buildkit/blob/master/cmd/buildkitd/main_unix.go#L51C1-L62 , which isn't being called...

Reply:
Sadly, the current `getLocalListener()` function is way to lax in terms of DACLs.

We need to parse the `--group` flag, resolve that to a SID, and use that SID to create a proper SDDL. We need to resolve the SID every time, because on Windows, the SID will always be different for each machine that was sysprepped. Also, if the machine is joined to an active directory, the group may be one belonging to AD, which will have a different SID compared to local groups. ie: you can have `localhost\Docker` and also `mydomain\Docker`.

The good news is that it should be easy to split the handling of the `--group` flag. If we omit it, we can default to none on Windows, and only allow the builtin `Administrators` group.

Reply:
> Sadly, the current getLocalListener() function is way to lax in terms of DACLs.

Just double checking; that's the implementation in BuildKit only, correct? Because that one's missing this part of the code (to apply the group); https://github.com/moby/moby/blob/faf84d7f0a1f2e6badff6f720a3e1e559c356fff/daemon/listeners/listeners_windows.go#L29-L35

Or is the code in Moby also incorrect, by applying the fixed  SID as starting point?;
https://github.com/moby/moby/blob/faf84d7f0a1f2e6badff6f720a3e1e559c356fff/daemon/listeners/listeners_windows.go#L26-L27

```go
// allow Administrators and SYSTEM, plus whatever additional users or groups were specified
sddl := "D:P(A;;GA;;;BA)(A;;GA;;;SY)"
```